woocommerce/wc-admin
# Fix reselecting advanced filters in the customer list

Fixes #4423

Reselecting advanced filters in the customer list (WooCommerce -> Customers) was clearing all selected filters. This fixes that by adding the advanced args to the config.

It also goes through the rest of the analytics screens that include an "Advanced" or "Comparison" option.

While testing this I noticed console errors regarding missing `key` values for loaded labels, this PR also makes sure there is a unique ID used as the key.

### Detailed test instructions:

- Open WooCommerce -> Customers
- Set up an advanced filter, make sure to press the "Filter" button to apply it
- Select the filter type dropdown and reselect "Advanced"
- Observe that the advanced filters remain
- Select the filter type dropdown and select a different filter type, then reselect Advanced
- Observe that the advanced filters remain
- Refresh the page
- Observe that the advanced filters remain
- There should be no JS errors in the console relating to missing `key` values

Also test:
- Analytics -> Orders -> Advanced
- Analytics -> Products -> Comparison
- Analytics -> Categories -> Comparison
- Analytics -> Coupons -> Comparison
- Analytics -> Taxes -> Comparison
- Analytics -> Downloads -> Advanced
